lib/Dialect/TritonNvidiaGPU/Transforms/TensorMemoryAllocation.cpp

```diff
@@ -176,8 +176,15 @@ static TMemChunk allocFirstFit(MemoryBitMap &memoryMap,
 
 static Operation *getAlloc(Value value) {
   Operation *op = value.getDefiningOp();
-  while (isa<triton::gpu::MemDescSubviewOp>(op)) {
-    op = op->getResult(0).getDefiningOp();
+  while (auto subOp = dyn_cast<triton::gpu::MemDescSubviewOp>(op)) {
+    if (subOp.getSrc().getDefiningOp()) {
+      op = subOp.getSrc().getDefiningOp();
+    } else {
+      auto arg = cast<BlockArgument>(subOp.getSrc());
+      auto wsOp = op->getParentOfType<triton::gpu::WarpSpecializeOp>();
+      auto capture = wsOp.getExplicitCaptures()[arg.getArgNumber()];
```

Comment:
you need to check or assert that the arg belongs to wsOp. Best would be to get wsOp from arg.getOwner()